lib/DBIx/Connector/Pg.pm

```diff
@@ -0,0 +1,60 @@
+package DBIx::Connector::Pg;
+# ABSTRACT: ...
+
+use strict;
+use warnings;
+use parent 'DBIx::Connector';
+
+our $VERSION = '0.001';
+
+sub connected {
+    my $self = shift;
+    return unless $self->_seems_connected;
+    my $dbh = $self->{_dbh} or return;
+    $dbh->state and $dbh->state =~ /^08|^57/ and return;
+    return 1;
+    #dbh->state and $dbh->state =~ /^BI/ and return 1;
+    #return $self->driver->ping($dbh);
+}
```

Comment:
When I first saw this I was tempted to suggest to uncomment line 17. But on a 2nd thought, I think this is more accurate.

Only, perhaps it's a good idea to treat the `57` error class or at least error number `57014` (query canceled) as `connected`. If a query has been cancelled we don't want it to be silently re-tried. And for the other errors in this class, it does not make sense to repeat them either. If the database has been dropped it's gone. If it has been shut down it's not going to be up again immediately afterwards.

So, maybe better to change line 14 to `$dbh->state and $dbh->state =~ /^08/ and return;`.